aws/resource_aws_greengrass_device_definition_test.go

```diff
@@ -0,0 +1,195 @@
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", fmt.Sprintf("device_definition_%s", rString)),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					resource.TestCheckResourceAttr("aws_greengrass_device_definition.test", "tags.tagKey", "tagValue"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSGreengrassDeviceDefinition_DefinitionVersion(t *testing.T) {
+	rString := acctest.RandString(8)
+	resourceName := "aws_greengrass_device_definition.test"
+
+	device := map[string]interface{}{
+		"sync_shadow": false,
+		"id":          "device_id",
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGreengrassDeviceDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSGreengrassDeviceDefinitionConfig_definitionVersion(rString),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", fmt.Sprintf("device_definition_%s", rString)),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					testAccCheckGreengrassDevice_checkDevice(resourceName, device),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckGreengrassDevice_checkDevice(n string, expectedDevice map[string]interface{}) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Greengrass Device Definition ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).greengrassconn
+
+		getDeviceInput := &greengrass.GetDeviceDefinitionInput{
+			DeviceDefinitionId: aws.String(rs.Primary.ID),
+		}
+		definitionOut, err := conn.GetDeviceDefinition(getDeviceInput)
+
+		if err != nil {
+			return err
+		}
+
+		getVersionInput := &greengrass.GetDeviceDefinitionVersionInput{
+			DeviceDefinitionId:        aws.String(rs.Primary.ID),
+			DeviceDefinitionVersionId: definitionOut.LatestVersion,
+		}
+		versionOut, err := conn.GetDeviceDefinitionVersion(getVersionInput)
+		if err != nil {
+			return err
+		}
+
+		device := versionOut.Definition.Devices[0]
+		expectedSyncShadow := expectedDevice["sync_shadow"].(bool)
+		if *device.SyncShadow != expectedSyncShadow {
+			return fmt.Errorf("Sync Shadow %t is not equal expected %t", *device.SyncShadow, expectedSyncShadow)
+		}
+
+		expectedDeviceId := expectedDevice["id"].(string)
+		if *device.Id != expectedDeviceId {
+			return fmt.Errorf("Device Id %s is not equal expected %s", *device.Id, expectedDeviceId)
+		}
+
+		expectedCertArn, err := getAttrFromResourceState("aws_iot_certificate.foo_cert", "arn", s)
```

Comment:
Could this be addressed by using the `TestCheckResourceAttrPair` function from https://godoc.org/github.com/hashicorp/terraform-plugin-sdk/helper/resource#TestCheckResourceAttrPair in the `TestStep`? For example `resource.TestCheckResourceAttrPair(resourceName, "device_definition_version.device.certificate_arn", "aws_iot_certificate.foo_cert", "arn")`.